contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -1517,10 +1517,12 @@ STRICT
 LANGUAGE SQL IMMUTABLE;
 
 CREATE OR REPLACE FUNCTION sys.space(IN number INTEGER, OUT result SYS.VARCHAR) AS $$
--- sys.varchar has default length of 1, so we have to pass in 'number' to be the
--- type modifier.
 BEGIN
-	EXECUTE pg_catalog.format(E'SELECT repeat(\' \', %s)::SYS.VARCHAR(%s)', number, number) INTO result;
+    IF number < 0 THEN
+        result := NULL;
+    ELSE
+        result := PG_CATALOG.repeat(' ',number);
```

Comment:
will it be able to handle INT_MAX?

Reply:
TSQL has a limitation of 8000 character spaces for SPACE function. Added the same support now.